### PR TITLE
Feature/ic2 45 cache unit test

### PR DESCRIPTION
**Pull request by dhlee-icon**
- Add Unit test to all caches
- Fix minor bugs in ActivePrepCache
- Change method name of GetSnapShot to Flush

### CODE REVIEW

**Comment by eunsoo-icon** [comment]
We need to talk about empty cache management.

Reset() : revert the changes
Flush() : write cache to DB

I think empty cache must deleted in Reset() and Flush()

Error case
1. TX1 delete Account A -> cache of Account A is empty
2. TX1 failed and cache.Reset() -> Account A remains empty
3. following TXs do not lookup Account A
4. cache.Flush() will delete Account A

is it right?

File: icon/iiss/icstate/accountCache.go [hunk 1]
```diff
@@ -70,15 +71,20 @@ func (c *AccountCache) Clear() {
...
-			} else {
-				account.Set(ToAccount(value.Object(), account.address))
-			}
+		if value == nil {
+			delete(c.accounts, key)
+		} else {
+			delete(c.accounts, key)
+			newac := newAccount(addr)
+			newac.Set(ToAccount(value.Object(), addr))
+			c.accounts[key] = newac
```
Comment (eunsoo-icon) [inline icon/iiss/icstate/accountCache.go:87]: Why delete cache? Any reason? You can overwrite it.

Comment (dhlee-icon) [reply icon/iiss/icstate/accountCache.go:87]: I tried to Set() account which is already in the cache.
But some values are distorted during copy value.
I deleted the item and create new Account and assigned, then it worked

Comment (dhlee-icon) [reply icon/iiss/icstate/accountCache.go:87]: account.Set(ToAccount(value.Object(), addr)) <= this doesn't work

Comment (dhlee-icon) [reply icon/iiss/icstate/accountCache.go]: I debugged account type and changed Set() part as you said. Thank you

**Comment by dhlee-icon** [comment]
I added PrepManage.go, removing BigIntZero

**Review by eunsoo-icon — APPROVED** [review]

**Pull request merged**